src/shared/components/oncoprint/controls/OncoprintControls.tsx

```diff
@@ -381,17 +404,118 @@ export default class OncoprintControls extends React.Component<IOncoprintControl
         }
     }
 
+    @autobind
+    private onChangeEditableDiv(text:string):string {
+
+        // get values from input string
+        const elements = this.splitTextField(text);
```

Comment:
I think this logic deserves unit tests - on how `text` is processed, and on `splitTextField`/which treatments are detected

Reply:
I fully agree. However, these methods are private and not accessible to any test that I write. Can you provide a suggestion on how to do this?

Reply:
The way I usually do this is pull it out into a helper function which we put in a file called, say, OncoprintControlsUtils.ts. That function is exported, and would take the minimum possible data as arguments, for ease of testing. So for example, it wouldn't take the whole OncoprintControls object, it would just take the value of `this._selectedTreatmentOptions`, `this.treatmentOptionsByValueMap`.

splitTextField is already a pure function so you could just move that into utils entirely and not put it as a class method.

Reply:
Extraction of helper functions was performed and unit test are included. Thanks for the push in the right direction :)